pkg/model/coordinator/milestones.go

```diff
@@ -45,20 +45,35 @@ func createMilestone(index milestone.Index, networkID uint64, parents hornet.Mes
 	}
 
 	iotaMsg := &iotago.Message{
-		NetworkID: networkID,
+		NetworkID: coo.networkID,
 		Parents:   parentsSliceOfArray,
 		Payload:   msPayload,
 	}
 
-	if err := msPayload.Sign(milestoneIndexSigner.SigningFunc()); err != nil {
+	signingFunc := milestoneIndexSigner.SigningFunc()
+	if err := msPayload.Sign(func(pubKeys []iotago.MilestonePublicKey, msEssence []byte) (sigs []iotago.MilestoneSignature, err error) {
+		for i := 0; i < coo.opts.signingRetryAmount; i++ {
```

Comment:
I should at least try once, even if the retry amount is set to 0.
```suggestion
		for i := 0; i <= coo.opts.signingRetryAmount; i++ {
```